NumberDisplay.java

```diff
@@ -54,9 +54,10 @@ public String getDisplayValue()
      */
     public void setValue(int replacementValue)
     {
-        if((replacementValue >= 0) && (replacementValue < limit)) {
+        if((replacementValue > 0) && (replacementValue < limit)) {
```

Comment:
This breaks number display's purpose. You will  never be able to set an even hour time now.